frontend/packages/console-shared/src/hooks/useK8sModel.ts

```diff
@@ -2,17 +2,21 @@
 // @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
 import { useSelector } from 'react-redux';
 import { UseK8sModel } from '@console/dynamic-plugin-sdk';
+import { transformGroupVersionKindToReference } from '@console/dynamic-plugin-sdk/src/utils/k8s/k8s-ref';
 import { K8sKind, kindForReference } from '@console/internal/module/k8s';
 import { RootState } from '@console/internal/redux';
 
 // Hook that retrieves the k8s model for provided groupVersionKind reference. Hook version of
 // `connectToModel`.
-export const useK8sModel: UseK8sModel = (groupVersionKind) => [
-  useSelector<RootState, K8sKind>(({ k8s }) =>
-    groupVersionKind
-      ? k8s.getIn(['RESOURCES', 'models', groupVersionKind]) ??
-        k8s.getIn(['RESOURCES', 'models', kindForReference(groupVersionKind)])
-      : undefined,
-  ),
-  useSelector<RootState, boolean>(({ k8s }) => k8s.getIn(['RESOURCES', 'inFlight']) ?? false),
-];
+export const useK8sModel: UseK8sModel = (groupVersionKind) => {
+  const kindReference = transformGroupVersionKindToReference(groupVersionKind);
```

Comment:
`transformGroupVersionKindToReference` should not be part of the API. It can be an internal utility

Reply:
by the internal utility, you mean `transformGroupVersionKindToReference`should not be part of SDK? as later we need to migrate `useK8sModel` as well to SDK so consuming `transformGroupVersionKindToReference` without exposing as core-api's

Reply:
yes, we can consume internally but do not expose. The discussion is the same as https://github.com/openshift/console/pull/10303/files#r742744802